torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4547,6 +4547,16 @@ def _tensor(shape, dtype=dtype, low=None, high=None):
 
     return [SampleInput(tensor, args=args) for tensor, args in test_cases]
 
+def sample_inputs_one_hot(op_info, device, dtype, requires_grad, **kwargs):
+    def make_tensor_(shape, *, low, high):
+        return make_tensor(shape, device=device, dtype=dtype, low=low, high=high, requires_grad=requires_grad)
+
+    return [
+        SampleInput(make_tensor_((S,), low=0, high=10)),
+        SampleInput(make_tensor_((S,), low=0, high=5), kwargs=dict(num_classes=10)),
+        SampleInput(make_tensor_((L, M, S), low=0, high=10)),
```

Comment:
For completeness, could we test a 3D tensor like this one but also have num_classes?